### ISSUE TITLE
Power BI Action: Publish - ##[error]Operation returned an invalid status code 'BadRequest'

### ISSUE THREAD

**Issue by PtolemaiosII** [maikvandergaag/msft-extensions#296]
Hi @maikvandergaag,

I use Power BI Action to publish/update existing .pbix file in my workspace. The release pipeline start to throw an error today when I do a new release (was working properly on 3 days ago on July 6th). Attached is the log from the Agent. Can you please help me on this?

![image](https://user-images.githubusercontent.com/20583343/125114097-5cb8c680-e09e-11eb-8504-a9d4a7278a15.png)

**Comment by maikvandergaag** [comment]
Could you add a variable System.debug value with the value true and then send the log

**Comment by PtolemaiosII** [comment]
Attached is the log file after adding variable System.Debug = true
[log.txt](https://github.com/maikvandergaag/msft-extensions/files/6793648/log.txt)

**Comment by maikvandergaag** [comment]
A new version is being  deployed to production as we speak. Could you please check that version?

**Comment by dimabutenko** [comment]
@maikvandergaag I have the same issue. Any updates?

**Comment by AliMzahem** [comment]
I have the same issue. My pbix file data source is Powerbi dataset. Publishing another file with source dataset from amazon redshift worked fine for me.

**Comment by naveedzaidi36** [comment]
Facing the same 400 BadRequest Error in UpdateGateway action.

**Comment by maikvandergaag** [comment]
@AliMzahem: Just saw your log file, could you try the following:
- Upload the report by hand, does that work?
- What steps do work. Is the workspace already there for example?
- Does it work when you remove the spaces in the names?

**Comment by AliMzahem** [comment]
@maikvandergaag
-  Uploading the report by hand works fine.
-  The workspace has been already created.
- I have tried having a name with no space for the file but it did not work.

**Comment by njais** [comment]
Hi, I have the same error message while trying to publish report to the workspace using Power Bi Actions via release pipeline. Below some findings: 

- Uploading the report by hand works fine.
- The workspace has been already created with service principle added as an admin to the workspace
- I have tried having a name with no space for the file but it did not work.
- The other reports are getting published fine via release pipeline and to the same workspace having datasources like excel,  tabular cube
- The pbix file which is getting this error message while publishing via release pipeline uses PowerBI dataset connection as a data source, so does this mean the extension has a limitation with datasource such as Power Bi dataset connection?

Could you please suggest/help as I am stuck for a while now..thanks

**Comment by maikvandergaag** [comment]
Agent configuration should not be an issue when all required components are installed on the server.

Did you try to create/import the dataset and let the process also create the workspace? And did you also try it with an empty report?

**Comment by dimabutenko** [comment]
Hi. Today I finished my investigations. I'm shure that problem not in this extension. 
1. I can open and publish(by PowerShell and Postman) report from my work PC.
2. I can copy report file to Build server and also can publish report by PowerShell and Postman.
3. But when I pull reports on Build server from github(we use LFS for reports) I got error with publishing. Also when I copy report pulled from github to my work PS I getting same error.

![image](https://user-images.githubusercontent.com/6714627/136946110-10b7a8fc-306e-4cc1-9d27-8e14a2059c0f.png)
![image](https://user-images.githubusercontent.com/6714627/136946133-feffebe0-40ea-4b07-b2d8-310678eaf788.png)


It looks like problem with git.
And I need to investigate in this place
